lib/connect.js

```diff
@@ -36,8 +36,9 @@ const CancelablePromise = require('./util/cancelablepromise');
 const Log = require('./util/log');
 const { validateBandwidthProfile } = require('./util/validate');
 
-const safariVersion = guessBrowser() === 'safari'
-  && navigator.userAgent.match(/Version\/([^\s]+)/)[1];
+const safariVersion = guessBrowser() === 'safari' && (
```

Comment:
@manjeshbhargav, 

We should add unit tests for this. I suggest move this into a seperate file (Perhaps into `test/lib/guessbrowser.js`) where it can be unit tested against offending user agent -  and export more structured version from there.   

looks like we have another problem that guessBrowser is returning 'safari' for this Chrome on iOS.
@manjeshbhargav - I think we should also file a task for consolidating our version checks, We have lot of them sprinkled around.

Reply:
@makarandp0 Pretty soon we will have to refactor our browser detection logic to no longer use `userAgent`, since it will be frozen and deprecated. So we can do this then.

Reply:
@manjeshbhargav - That's a good point, but  u/a change is chrome only (for now) and it's many months away. Also this work of consolidating the version check will also prepare us for eventual removal of useragent. Once we are done with this consolidation user agent removal will be much simpler localized change.